Automattic/jetpack
# Update BlockStylesSelector component and implement in Eventbrite

This makes some changes to the BlockStylesSelector component and implements it in the Eventbrite block.

Fixes #14527

#### Changes proposed in this Pull Request:

* The preview for the button Embed Type on the EventBrite block will now be a real block preview.
* The EventBrite block also now allows you to change the Embed Type from the BlockControls.
* The `useModal` attribute in the block is now deprecated and replaced with a `style` attribute.
* The component now also includes the inspector controls and creates a the block toolbar

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* adding to existing features

#### Testing instructions:






* Add an Eventbrite block
* Add an event
* Check that you can still toggle between a button and an inline embed
<img width="290" alt="Screenshot 2020-01-31 at 13 47 00" src="https://user-images.githubusercontent.com/275961/73544273-bdb38800-4430-11ea-8247-e8d79bd0d636.png">

* Check that the Calendly, OpenTable and EventBrite blocks all have a Toolbar which allows you to change the embed style:
<img width="290" alt="Screenshot 2020-01-31 at 14 16 42" src="https://user-images.githubusercontent.com/275961/73546040-69120c00-4434-11ea-88cd-fba3a82f54b0.png">

* You also need to test the deprecation for old Eventbrite blocks using the old attributes.
* Create an Eventbrite block using master
* Set the Embed type to be "Modal"
* Save the post
* Check out this branch
* Reload the editor
* Check that the block is still in modal type
* Open the editor in code view and check you see `"style":"modal"` in the block attributes.

#### Proposed changelog entry for your changes:

* no changelog
